daemon/graphdriver/driver.go

```diff
@@ -167,30 +176,23 @@ func New(root string, options []string) (driver Driver, err error) {
 }
 
 // scanPriorDrivers returns an un-ordered scan of directories of prior storage drivers
-func scanPriorDrivers(root string) []string {
-	priorDrivers := []string{}
+func scanPriorDrivers(root string) map[string]string {
+	priorDrivers := map[string]string{}
 	for driver := range drivers {
 		p := filepath.Join(root, driver)
 		if _, err := os.Stat(p); err == nil {
-			priorDrivers = append(priorDrivers, driver)
+			priorDrivers[driver] = p
 		}
 	}
 	return priorDrivers
 }
 
-func checkPriorDriver(name, root string) error {
-	priorDrivers := []string{}
-	for _, prior := range scanPriorDrivers(root) {
-		if prior != name && prior != "vfs" {
-			if _, err := os.Stat(filepath.Join(root, prior)); err == nil {
-				priorDrivers = append(priorDrivers, prior)
-			}
-		}
+func multipleDriversError(root string, drivers map[string]string) error {
+	names := make([]string, len(drivers))
```

Comment:
I think it's better to use `var names []string` and `append` here for readability.